### ISSUE TITLE
Unstyled  hrs

### ISSUE THREAD

**Issue by elinw** [Joomla3-Admin-template/joomla-cms#60]
A lot of layouts have unstyled hrs at the bottom of different sections, and I know people are going to go crazy trying to style those and/or make them invisible. On the featured article layout there is class="divider-vertical" ... could we put that everywhere the hr is used?

**Comment by nternetinspired** [comment]
Actually, I think <hr> is preferable as it contains semantic value. 

However, the defined usage of <hr> has changed with the html5 spec. It is now used to represent a thematic break in content, so it's use a styling element is incorrect. Current spec is here: http://dev.w3.org/html5/markup/hr.html

Probably a better review, with clear examples, is: http://html5doctor.com/small-hr-element/

**Comment by nternetinspired** [comment]
My interpretation of the spec is that the current use of hr in Protostar is semantically incorrect.

**Comment by elinw** [comment]
It' not in Protostar that it is the issue it is in the core outputs. It is
important to keep that distinction clear. A designer of a template can do
as he or she wants.

**Comment by nternetinspired** [comment]
Well I've only noticed in Protostar.

The semantics argument remains though. Wherever hr is used, it should be used correctly and in line with html specs.

Moreover, if there is an element that carries semantic value that can be used for a particular purpose it should always be used in preference to non-semantic element (e.g. <div class="divider-vertical".. ).

In relation to this issue, whether hr is the correct element to use or not depends on context. I'll take a look at the featured layout now...

**Comment by nternetinspired** [comment]
This does also relate to the template in use, as the doctype is often significant in determining the correct usage of a particular element. hr is a great example as the spec definition for that element has changed between HTML4 and html.

**Comment by nternetinspired** [comment]
In the case of the featured view I would say that, if the doctype is not html (i.e. html5) then the markup is simply extraneous, it could be removed entirely and the styling applied instead to the parent element (keeping a nice separation of function and style). If the doctype is html then the element is OK if the new section element is not going to be used.

**Comment by nternetinspired** [comment]
Maybe a better option would be to simply remove all hr elements from core views, leaving it up to the front-end template developer to decide how to separate and style content?

Personally, I don't like to see a mixing of style and function like this.

**Comment by elinw** [comment]
Yes would you make a pull request to remove them?  If they are style elements they do not belong in the core layouts.

**Comment by betweenbrain** [comment]
Agreed! The `<hr>` in core is only being used by Bootstrap templates for style purposes only. They don't even need to be used anywhere as the Sam CSS can be applied to the article elements.

We also need to get those empty clearing `<div>` out too.

**Comment by rdeutz** [comment]
Why we are not using the output from 2.5, there was a good semantic markup Angie created over the years in beez. Starting over again makes not so much sense to me.

**Comment by betweenbrain** [comment]
Robert,

I had an exchange with Kyle on Twitter about this last night. Basically,
the core markup changes are being driven by Bootstrap's CSS.
I personally feel that this is absolutely the wrong way to approach this
<strong>IF</strong> the goal is to have clean semantic markup. See
https://twitter.com/kyleledbetter/status/238822752138706945 and
https://twitter.com/Arlen/status/238833863202328577 for some insight into
our exchanges.

Kyle clarified that this is the new paradigm for using the core
boostrap/jui.

I do agree with you that starting over doesn't make much sense as we could
continue with greatly improving the markup from 2.5.

In any case, I'd be happy to make a pull request tonight to remove
the unnecessary `<hr>` elements if no one else has already done it.

Best,

Matt Thomas
Founder betweenbrain http://betweenbrain.com/™
Lead Developer Construct Template Development
Frameworkhttp://construct-framework.com/
Phone: 203.632.9322
Twitter: @betweenbrain
Github: https://github.com/betweenbrain

**Comment by rdeutz** [comment]
yeah I have see this discussion, looks for me that this is using markup to create a visual design. To be honest I think that is the wrong way

Cheers,
Robert

**Comment by nternetinspired** [comment]
Agreed. This goes clearly against web standards best practice. The markup should be completely free of stylistic elements.

I don't see any such stylistic markup elements in the Twitter Bootstrap repo...

**Comment by betweenbrain** [comment]
Pull request at https://github.com/joomla/joomla-cms/pull/330

**Comment by elinw** [comment]
These were actually removed already. There are some other semantic markup
elements that are being worked on, please see the CMS tracker category
templates. Testers much appreciated.

Elin
